legacy/src/main/java/org/opensearch/sql/legacy/executor/format/PrettyFormatRestExecutor.java

```diff
@@ -134,7 +134,9 @@ private Protocol buildProtocolForDefaultQuery(Client client, DefaultQueryAction
 
   private boolean isDefaultCursor(SearchResponse searchResponse, DefaultQueryAction queryAction) {
     if (LocalClusterState.state().getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER)) {
-      if (searchResponse.getHits().getTotalHits().value < queryAction.getSqlRequest().fetchSize()) {
+      if (queryAction.getSqlRequest().fetchSize() == 0
+          || searchResponse.getHits().getTotalHits().value
+              < queryAction.getSqlRequest().fetchSize()) {
```

Comment:
No test cover this?

Reply:
I know we have IT to cover legacy part of the code but I don't see existing test file to cover UT for this class. I've created one and added UT for this function.

Reply:
Thanks for adding the test! Could you test API behavior instead of internal impl in a private method? I see execute API with `DefaultQueryAction` can cover your changes?

Reply:
Sure, I need to get this one merged and can look into this as follow-up.